### ISSUE TITLE
Thoughts on improvements to the drawing assistants

### ISSUE THREAD

**Issue by DarrenTAnims** [opentoonz/opentoonz#5154]
### Is your feature request related to a problem? Please describe

Firstly, I want to say again, the drawing assistants are soooooooo useful. They're so much more feature-rich than I expected and it was a pleasant surprise when the features started appearing. But after trialling them all for a while, I have a few thoughts about small usability issues that could help. You may have already thought of these, or not, but feel free to ignore them all, but if you've got time to consider them @blackwarthog, then I'd appreciate it.

### Describe the solution you'd like

1. Selecting a guide, after moving from one frame to another, is tricky and you often move them when trying to select. What would help would be a dropdown on the options bar listing all of the guides visible on the current frame.

2. On the 2 "Replicator" guides, instead of having a option on the toolbar "Discrete Angle", could we not use a keyboard modifier like we do when using the brush tool's straight line feature? So if you was hold the Ctrl key while rotating your guide (e.g. the Replicator Mirror), it would snap at discrete angles.
  I also notice that the angles vary when using the replicator star, depending on the number of mirrors.

3. Also, the discrete angles for the Mirror are 45 degrees. I think it would eb helpful to have the angle set in an option, either on the options bar or in the preferences. I really like the 15 degree angles that Tahoma uses, but I suspect that not everyone does, so an option could be best.

4. If we did add a modifier key to lock to discrete angles, it'd be great to use it with some of the other controls. e.g. when adjusting the filled widgets of the Perspective guide.

5. What do "Skip first Tracks" & "Skip last tracks" mean?
   There's nothing in the docs yet and experimenting didn't help me to understand.
   
6. What does the "Order" option do?

7. On most of the guides, you can just click and drag the widget with the plus in it to move the whole guide. However, with the line & vanishing point guides you have to hold the Shift key. For consistency, could those 2 also move when you drag the plus widget too? And maybe assign a modifier key to move just that one point separately?

**Comment by RattusaurusRex** [comment]
For #5, this is mentioned in one of the PRs: Also i've added two new options to each replicator and to the jitter: "skip first track" and "skip last tracks". These options allow to exclude some tracks from replication/jittering.

However, I'm still not sure what that means.  If I use the skip tracks at all, it just seems to stop the replication assistants from working altogether.  The only one it seems to work on is Jitter, which I tested with a Replicator + Jitter Assistant.

**Comment by DarrenTAnims** [comment]
Yes @RattusaurusRex, I saw that in a PR, but what's a track? Is that a column, a frame, an assistant, something else?

**Comment by RattusaurusRex** [comment]
@DarrenTAnims I think a track is the path of the brush.  If using one of the Replicators to replicate the brush 10 times, you would have 10 tracks, for example.

**Comment by blackwarthog** [comment]
1, 2, 3, 4. I'll try to implement it.

5. Yes, tracks are paths of a brush. When you've applied a replicator you will have multiple brush paths (tracks). And the following replicator will take multiple tracks to input. Fields "skip first/last tracks" allow you to apply a replication not for all input tracks, but only for part of them. For example you can apply jittering only for single row of the grid.

6. The "order" field allows you to control the order of applying the replicators. You may apply a grid and next apply the mirror, or use other sequence: mirror, then grid. And the results will be different. So order in sequence of applying modifiers is significant. You may see it in this video since 0:40:

https://user-images.githubusercontent.com/5143743/263763575-7f91065e-b73c-4e4c-be58-3e4205c66c61.mp4

7. As i remember i've fixed the behavior of the "plus" handle of the line assistant. But for the vanishing point i'm not sure. I think that the moving the vanishing point (plus) with keeping other points at place are useful. Additional points may be linked to some geometry.

**Comment by blackwarthog** [comment]
> I also notice that the angles vary when using the replicator star, depending on the number of mirrors.

Yes, currently the discrete step is 1/8 of the angle between the star rays (instead of 1/8 of 360). I think it is more useful.

**Comment by DarrenTAnims** [comment]
Awesome. Thanks for your reply my friend.
1, 2 ,3, 4 - amazing!
5. Fantastic! That makes a lot of sense. So it's when combining one guide with another.
6. That video explains it perfectly. I missed that part of the demo the last time I watched. Perfect. Thank you.
7. Yes, I remember using the old vanishing point tool and found it useful to keep the other points in place, but that's where holding a key like Ctrl would be helpful. I think (and I could be on my own here, so this isn't gospel for all) that more often you'd want to move the whole guide. Plus, as I say, it would keep it consistent with the rest. But I'll say not more. You've done a brilliant job with this. I honestly am so happy about them all 😀

8. One last thing I forgot to mention in this post, but had in another. You said you was going to add in a new command to the Level menu and to the timeline toolbar to add a guide level. And while you was there, if possible, one to add a note level? I'm really trying hard to not sound demanding, but if you have time ... pretty please 🤞😊🤞

Thanks again!

**Comment by RattusaurusRex** [comment]
@DarrenTAnims There's already one for a New Note Level: 
![image](https://github.com/opentoonz/opentoonz/assets/60277069/3198424d-b4cc-41cb-84e6-a35bf19f26ed)
![image](https://github.com/opentoonz/opentoonz/assets/60277069/b0c75b83-8b25-479e-9272-0127a9fc9076)

@blackwarthog I haven't used the Assistants very much yet, but I'm liking them a lot so far.  One thing I noticed while using the Mirror Assistant was that I couldn't use the Eraser with it.  So even though I only had to draw something once, I had to erase mistakes twice.  Could the Eraser use the Assistants?

Another thing I tried was to use the Assistants to create a visual grid on the canvas (without actually having the Assistants affect the brush).  Do the cursor guide lines need to be shown when the Assistants checkbox is not checked? 
![image](https://github.com/opentoonz/opentoonz/assets/60277069/946e6bb6-db99-4eb8-b7a8-1f4e611f7649)

**Comment by blackwarthog** [comment]
I'll try to implement assistants for eraser too.

Visible "inactive" guidelines is useful. They helps to choose exact points between grid lines, especially in perspective where we have no ability to cover all dimensions by grids. Simple example: by two painted diagonal verticies of a rectangle you may find the remained two verts by using cursor guidelines.

In the both videos in this post (https://github.com/opentoonz/opentoonz/pull/5100#event-10542564212) i've used the cursor-hovering assistants to determine the proper points where begin and end lines. In this videos assistants are enabled, but if you want to draw free (non-straight) strokes this helpers is the only way to make proper projection.
